framework/CodeInterfaces/SERPENT/SerpentInterface.py

```diff
@@ -0,0 +1,191 @@
+      @ In, fileList, the string list of filenames to pick from.
+      @ Out, ext, the string extension that the desired filename ends with.
+      """
+      found = False
+      for index,inputFile in enumerate(fileList):
+        if inputFile.getExt() == ext:
+          found=True
+          break
+      if not found:
+        raise IOError('No InputFile with extension '+ext+' found!')
+      return index,inputFile
+
+    #prepend
+    todo = ''
+    todo += clargs['pre']+' '
+    todo += executable
+    index=None
+    #inputs
+    for flag,exts in clargs['input'].items():
+      if flag == 'noarg':
+        for ext in exts:
+          idx,fname = getFileWithExtension(inputFiles,ext.strip('.'))
+          todo+=' '+fname.getFilename()
+          if index == None:
+            index = idx
+        continue
+      todo += ' '+flag
+      for ext in exts:
+        idx,fname = getFileWithExtension(inputFiles,ext.strip('.'))
+        todo+=' '+fname.getFilename()
+        if index == None:
+          index = idx
+    #outputs
+    #FIXME I think if you give multiple output flags this could result in overwriting
+    self.caseName = inputFiles[index].getBase()
+    outfile = 'out~'+self.caseName
+    if 'output' in clargs.keys():
+      todo+=' '+clargs['output']+' '+outfile
+    #text flags
+    todo+=' '+clargs['text']
+    #postpend
+    todo+=' '+clargs['post']
+    returnCommand = [('parallel',todo)],outfile
+    print('Execution Command: '+str(returnCommand[0]))
+    return returnCommand
+
+  def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
+    """
+      This method is used to generate an input based on the information passed in.
+      @ In, currentInputFiles, list,  list of current input files (input files from last this method call)
+      @ In, oriInputFiles, list, list of the original input files
+      @ In, samplerType, string, Sampler type (e.g. MonteCarlo, Adaptive, etc. see manual Samplers section)
+      @ In, Kwargs, dictionary, kwarded dictionary of parameters. In this dictionary there is another dictionary called "SampledVars"
+            where RAVEN stores the variables that got sampled (e.g. Kwargs['SampledVars'] => {'var1':10,'var2':40})
+      @ Out, newInputFiles, list, list of newer input files, list of the new input files (modified and not)
+    """
+    indexes=[]
+    infiles=[]
+    origfiles=[]
+    #FIXME possible danger here from reading binary files
+    for index,inputFile in enumerate(currentInputFiles):
+      if inputFile.getExt() in self.getInputExtension():
+        indexes.append(index)
+        infiles.append(inputFile)
+    for index,inputFile in enumerate(origInputFiles):
+      if inputFile.getExt() in self.getInputExtension():
+        origfiles.append(inputFile)
+    parser = GenericParser.GenericParser(infiles)
+    parser.modifyInternalDictionary(**Kwargs)
+    parser.writeNewInput(currentInputFiles,origfiles)
+    return currentInputFiles
```

Comment:
this is copied/pasted by the Generic Code interface...

Can't' you just inherit from ```GenericCode```?